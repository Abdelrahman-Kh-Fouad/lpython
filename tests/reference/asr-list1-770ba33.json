{
    "basename": "asr-list1-770ba33",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/list1.py",
    "infile_hash": "0b28eb905e0a7b2b4656d1aa6a43a844be4a8f4764712a5fea66c32d",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-list1-770ba33.stdout",
    "stdout_hash": "2e94ad9aa2d6c65274c7b74296efe4380c9516b2e02657ae92108631",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}