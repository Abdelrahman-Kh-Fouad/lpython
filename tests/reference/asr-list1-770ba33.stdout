(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_List: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (List (Integer 4 [])) Source Public Required .false.), b: (Variable 2 b Local () () Default (List (Character 1 -2 () [])) Source Public Required .false.), c: (Variable 2 c Local () () Default (List (List (Integer 4 []))) Source Public Required .false.), d: (Variable 2 d Local () () Default (Integer 4 []) Source Public Required .false.), e: (Variable 2 e Local () () Default (List (List (Character 1 -2 () []))) Source Public Required .false.)}) test_List [] [(= (Var 2 a) (ConstantList [(ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])) (ConstantInteger 3 (Integer 4 []))] (List (Integer 4 []))) ()) (= (Var 2 a) (ConstantList [(UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 []))) (UnaryOp USub (ConstantInteger 2 (Integer 4 [])) (Integer 4 []) (ConstantInteger -2 (Integer 4 []))) (UnaryOp USub (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) (ConstantInteger -1 (Integer 4 [])))] (List (Integer 4 []))) ()) (= (Var 2 b) (ConstantList [(ConstantString "a" (Character 1 1 () [])) (ConstantString "b" (Character 1 1 () [])) (ConstantString "c" (Character 1 1 () []))] (List (Character 1 1 () []))) ()) (= (Var 2 c) (ConstantList [(ConstantList [(ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])) (ConstantInteger 3 (Integer 4 []))] (List (Integer 4 []))) (ConstantList [(ConstantInteger 4 (Integer 4 [])) (ConstantInteger 5 (Integer 4 [])) (ConstantInteger 6 (Integer 4 []))] (List (Integer 4 [])))] (List (List (Integer 4 [])))) ()) (= (Var 2 d) (ArrayRef 2 a [(() (BinOp (ConstantInteger 2 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Integer 4 []) ()) ()) (= (Var 2 e) (ConstantList [(ConstantList [(ConstantString "a" (Character 1 1 () [])) (ConstantString "b" (Character 1 1 () [])) (ConstantString "c" (Character 1 1 () []))] (List (Character 1 1 () []))) (ConstantList [(ConstantString "d" (Character 1 1 () [])) (ConstantString "e" (Character 1 1 () []))] (List (Character 1 1 () [])))] (List (List (Character 1 1 () [])))) ())] Source Public Implementation () .false. .false.)}) [])
